services/retention/service.go

```diff
@@ -119,16 +118,14 @@ func (s *Service) deleteShards() {
 				}
 			}
 
-			for _, id := range s.TSDBStore.ShardIDs() {
-				if di, ok := deletedShardIDs[id]; ok {
-					if err := s.TSDBStore.DeleteShard(id); err != nil {
-						s.logger.Printf("failed to delete shard ID %d from database %s, retention policy %s: %s",
-							id, di.db, di.rp, err.Error())
-						continue
-					}
-					s.logger.Printf("shard ID %d from database %s, retention policy %s, deleted",
-						id, di.db, di.rp)
+			for id, info := range deletedShardIDs {
```

Comment:
I think this logic purposely iterated over the shard IDs to account for failed shard deletions so they could be retried.  This could leave orphan'd shards I think.